extensions-contrib/druid-iceberg-extensions/src/main/java/org/apache/druid/iceberg/input/IcebergCatalog.java

```diff
@@ -0,0 +1,104 @@
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.iceberg.input;
+
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import org.apache.druid.data.input.InputFormat;
+import org.apache.druid.iceberg.filter.IcebergFilter;
+import org.apache.druid.java.util.common.IAE;
+import org.apache.druid.java.util.common.RE;
+import org.apache.druid.java.util.common.logger.Logger;
+import org.apache.iceberg.BaseMetastoreCatalog;
+import org.apache.iceberg.FileScanTask;
+import org.apache.iceberg.TableScan;
+import org.apache.iceberg.catalog.Catalog;
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.io.CloseableIterable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/*
+ * Druid wrapper for an iceberg catalog.
+ * The configured catalog is used to load the specified iceberg table and retrieve the underlying live data files upto the latest snapshot.
+ * This does not perform any projections on the table yet, therefore all the underlying columns will be retrieved from the data files.
+ */
+
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = InputFormat.TYPE_PROPERTY)
+public abstract class IcebergCatalog
+{
+  private static final Logger log = new Logger(IcebergCatalog.class);
+
+  public abstract BaseMetastoreCatalog retrieveCatalog();
+
+  /**
+   * Extract the iceberg data files upto the latest snapshot associated with the table
+   *
+   * @param tableNamespace The catalog namespace under which the table is defined
+   * @param tableName      The iceberg table name
+   * @return a list of data file paths
+   */
+  public List<String> extractSnapshotDataFiles(
+      String tableNamespace,
+      String tableName,
+      IcebergFilter icebergFilter
+  )
+  {
+    Catalog catalog = retrieveCatalog();
+    Namespace namespace = Namespace.of(tableNamespace);
+    String tableIdentifier = tableNamespace + "." + tableName;
+
+    List<String> dataFilePaths = new ArrayList<>();
+
+    ClassLoader currCtxClassloader = Thread.currentThread().getContextClassLoader();
+    try {
+      Thread.currentThread().setContextClassLoader(getClass().getClassLoader());
+      TableIdentifier icebergTableIdentifier = catalog.listTables(namespace).stream()
+                                                      .filter(tableId -> tableId.toString().equals(tableIdentifier))
+                                                      .findFirst()
+                                                      .orElseThrow(() -> new IAE(
+                                                          " Couldn't retrieve table identifier for '%s'",
```

Comment:
is there an action that the user can take to remedy this error? Like verifying that the table indeed exists in the catalog.

Reply:
Updated the error message.